modules/taskFunctions.js

```diff
@@ -0,0 +1,108 @@
+class TaskCollection {
+  constructor() {
+    this.listContainer = document.querySelector('.tasks-container');
+    this.description = document.querySelector('#addDo');
+    this.completed = false;
+    this.taskIndex = 1;
+    this.toDoList = [];
+    this.editToDo = -1;
+  }
+
+  storedLocal() {
+    if (localStorage.getItem('tasks') == null) {
+      localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+    }
+
+    const store = localStorage.getItem('tasks');
+    if (store) {
+      this.toDoList = JSON.parse(store);
+    }
+  }
+
+  removeTask(index) {
+    const passIndex = index - 1;
+    this.toDoList = this.toDoList.filter(
+      (task, taskIndex) => taskIndex !== passIndex,
+    );
+    localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+    for (let i = 0; i < this.toDoList.length; i += 1) {
+      this.toDoList[i].index = i;
+    }
+
+    this.displayTasks();
+  }
+
+  removeAll() {
+    this.toDoList = this.toDoList.filter(() => false);
+    localStorage.setItem('tasks', JSON.stringify(this.toDoList));
+  }
+
+  completeTask(todoId) {
+    todoId -= 1;
+    this.toDoList = this.toDoList.map((task, index) => ({
+      ...task,
+      completedValue: index === todoId ? !task.completed : this.completed,
+    }));
+
+    this.displayTasks();
+  }
```

Comment:
- [ ] Great job working on this milestone.
- When I click on the checkbox to make task completed, the status does not change in the localstorage to completed:True.
- Secondly, I cannot tick the checkbox for more that one task. When I click the second, the first deselects.
- Finally when I click on `Clear all completed` even the tasks where the checkbox is not ticked are deleted. It is not supposed function like this.
![image](https://github.com/sulaimonaa/to-do-list/assets/62849872/9e758ac4-6b28-4afd-8b83-ba933acc14ed)

All these as per project requirements are supposed to work. Kindly look once one into these features